bigtable_input.py

```diff
@@ -45,22 +47,66 @@ def get_game_range_row_names(game_begin, game_end):
     Sample row name:
       g_0000000001_m001
 
-    To capture the range of all moves in the two given games,
-    the end row will need to go up to g_00..(N+1).
-
     Args:
       game_begin:  an integer of the beginning game number.
-      game_end:  an integer of the ending game number, inclusive.
+      game_end:  an integer of the ending game number, exclusive.
 
     Returns:
       The two string row numbers to pass to Bigtable as the row range.
     """
     row_fmt = 'g_{:0>10}_'
-    return row_fmt.format(game_begin), row_fmt.format(game_end + 1)
+    return row_fmt.format(game_begin), row_fmt.format(game_end)
+
+
+def make_single_array(ds, batch_size=8*1024):
+    full_result = None
+    with tf.Session() as sess:
+        ds = ds.batch(batch_size)
+        iterator = ds.make_initializable_iterator()
```

Comment:
why initializable_iterator vs make_one_shot_iterator()

Reply:
Unfortunately, this is a requirement for Bigtable iterators. Attempting to create a one-shot iterator of a Dataset constructed from a Bigtable yields the following error:

Failed to create a one-shot iterator for a dataset. `Dataset.make_one_shot_iterator()` does not support datasets that capture stateful objects, such as a `Variable` or `LookupTable`. In these cases, use `Dataset.make_initializable_iterator()`